nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -213,6 +213,146 @@ public void nullableAssignmentParameterArray() {
         .doTest();
   }
 
+  @Test
+  public void arraySubtyping() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static void test(@Nullable Integer[] nullableIntArr, Integer[] nonnullIntArr) {",
+            "    // legal",
+            "    Integer[] x1 = nonnullIntArr;",
+            "    // legal",
+            "    @Nullable Integer[] x2 = nullableIntArr;",
+            "    // legal (covariant array subtypes)",
+            "    x2 = nonnullIntArr;",
+            "    // BUG: Diagnostic contains: Cannot assign from type @Nullable Integer[] to type Integer[]",
+            "    x1 = nullableIntArr;",
```

Comment:
To be clear, we don't try to infer nullability annotations for elements of local variables of array type or nullability of generics for object types, right? Or is it just that once assigned a `@NonNull Integer[]` then `x1` must be assume to be incompatible with `@Nullable Integer[]`?

Reply:
We don't do any inference for array contents nullability.  So when we declare `Integer[] x1 = ...` that means we are declaring that the array element type of `x1` is `@NonNull`.